### DISCUSSION TITLE
Proper naming

### DISCUSSION THREAD

**Discussion by fractalf** [usebruno/bruno#383] [Ideas]
I get that each request is saved as a file and that `https://foo.bar/api` is a bad filename.

*But*, can you not have a `name` property in the .bru file that allowes for urls as a request name?

This is ugly, I would rather it was named exactly as I choose. 
There is no reason to link name <--> filename

Thanks :)

![image](https://github.com/usebruno/bruno/assets/5707457/751193e3-1ca4-41cb-9eda-9ecf89ff478f)

**Comment by fractalf** [comment]
I had to double check this, since this was an import from insomnia, but surely the name is "forced" into the same as the filename..

![image](https://github.com/usebruno/bruno/assets/5707457/6f6b44c5-d2ca-47d4-8252-1fe062333110)

Looking at the .bru file I see no reason for it to be like this, in fact I changed in the config file and it works just fine

![image](https://github.com/usebruno/bruno/assets/5707457/8104eef1-ac53-4f45-b565-ffc2b202979c)

**Comment by wouter-leistra** [comment]
I would like to add my support for proper naming... I don't care too much about what the filenames are, but i also use / and : characters in my api request names and found that we can update the name attribute in the meta object on the bru file. However if you now use rename on the bruno app it will throw an error and I also doubt if the name is used properly when sorting things.

I really hope this could be fixed.

**Comment by rcmaples** [comment]
+1 to this. 

I saw a couple of related issues (e.g. #219) and I was curious, is it not possible to translate system reserved characters into dashes for the file name and use the URL formatted name as display-only meta data when naming a file? I'm a bit out of my depth working on a project like this, but it seems doable 🤷

- User enters a name that can't be used by the filesystem: `/users/:user/repos`
- Bruno writes to disk a file called `-users--user-repos.bru` (maybe strip that leading slash to be `users--user-repos.bru`)
- Bruno maintains the desired name in the bru file:

```
# users--user-repos.bru
meta {
  name: /users/:user/repo
  type: http
  seq: 2
}

get {
  url: https://api.github.com/users/USERNAME/repos
  body: none
  auth: none
}
```

Looking at it now, I wonder if it would make sense to convert all special characters to dashes for the file name and leave the entry's display name as typed. Creating a request called `bob's great endpoint` in the UI creates `bob's great endpoint.bru`, but to do anything with it in the terminal, you've got to escape the spaces and `'` anyway  -->  `$ nano bob\'s\ great\ endpoint.bru`

I'd like to see that type of conversion, but for now I can keep changing the name in the bru files.
